validation/src/main/scala/hmda/validation/rules/lar/quality/Q022.scala

```diff
@@ -11,18 +11,22 @@ import scala.util.Try
 object Q022 {
   def apply(lar: LoanApplicationRegister, year: Int): Result = {
     // This edit should not fail if applicationDate is equal to "NA" or is otherwise non-convertible
-    val applicationYear = Try(lar.loan.applicationDate.substring(0, 4).toInt).getOrElse(year)
+    val applicationYear = parseYear(lar).getOrElse(year)
```

Comment:
The second method has the year passed in as a `Future`, so we won't know what year to pass to the `getOrElse()` method until the `Future` resolves.

I could call the `parseYear` function in the scope of the map function, but it seemed harder to read that way.